tune-threads/tune-threads.lisp

```diff
@@ -26,14 +26,12 @@
   "Time the quil program in qvm Q and return the number of seconds per thread per trial."
   (/ (time-quil-prog-raw q :num-trials num-trials :num-threads num-threads) (* num-threads num-trials)))
 
-(defun find-num-trials (q num-threads time-limit &optional (hint 0))
+(defun find-num-trials (q num-threads time-limit)
   "Choose the number of trials for the qvm/program Q that take around TIME-LIMIT seconds to complete."
-  (if (= hint 1)
-      1
-      (loop :for num-trials := 1 :then (* num-trials 2)
-            :for one-time := 0.0 :then (time-quil-prog-raw q :num-threads num-threads :num-trials num-trials)
-            :while (< one-time 0.1) ; avoid divide by zero and times very close to zero
-            :finally (return (1+ (floor (/ time-limit (/ one-time num-trials))))))))
+  (loop :for num-trials := 1 :then (* num-trials 2)
+        :for one-time := 0.0 :then (time-quil-prog-raw q :num-threads num-threads :num-trials num-trials)
+        :while (< one-time 0.1) ; avoid divide by zero and times very close to zero
```

Comment:
I'm confused about why `one-time` gets initialized to `0.0`. Doesn't this mean the first pass through the loop is a no-op?

Also, I think this means that `time-quil-prog-raw` never gets executed with `num-trials` set to `1`.

As I understand it:

```
num-trials := 1
one-time := 0.0
:while (< one-time 0.1) ; => t
;; back to top of loop
num-trials := (* num-trials 2) ; => 2
one-time := (time-quil-prog-raw ...)
```

If the intention is not to skip `num-trials = 1`, then you could instead

```lisp
(loop :for num-trials := 1 :then (* num-trials 2)
      :for one-time := (time-quil-prog-raw q :num-threads num-threads :num-trials num-trials)
   ...)
```

I know this is partly my fault for suggesting you move these variables into the `loop` clauses. I think the above change gets you back to your original loop semantics, before I started meddling :-)

Otherwise, if you *do* want to skip `num-trials = 1`, you could make the above change, plus directly initialize `num-trials := 2` and save a loop iteration.

Reply:
After reading the last comments, I think this code should be change in one of two directions. One is to allow timing to be less efficient by ignoring previous results. This has the advantage of reducing code complexity. The other is retain the level of efficiency that I have been aiming for, and to reorganize the code around a couple of structs. I am coding the latter now.